wolfcrypt/src/ecc.c

```diff
@@ -4036,6 +4130,16 @@ int wc_ecc_verify_hash_ex(mp_int *r, mp_int *s, const byte* hash,
 
 #else
 
+  /* checking if private key with no public part */
+  if (key->type == ECC_PRIVATEKEY_ONLY) {
+      WOLFSSL_MSG("Verify called with private key, generating public part");
+      err = wc_ecc_make_pub_ex(key, NULL, NULL);
+      if (err != MP_OKAY) {
+           WOLFSSL_MSG("Unable to extract public key");
+           return BAD_FUNC_ARG;
```

Comment:
Why wouldn't we return err?  Either this function supports calling with a private key only or it doesn't.  It seems odd to return bad caller function argument if that is a valid function argument.